src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -62,6 +62,30 @@ def __init_subclass__(cls, exponent_unit_map: dict[int, str]) -> None:
         cls._exponent_unit_map = exponent_unit_map
         super().__init_subclass__()
 
+    _zero_cache: dict[type, Quantity] = {}
+    """Cache for zero singletons.
+
+    This is a workaround for mypy getting confused when using @functools.cache and
+    @classmethod combined with returning Self. It believes the resulting type of this
+    method is Self and complains that members of the actual class don't exist in Self,
+    so we need to implement the cache ourselves.
+    """
+
+    @classmethod
+    def zero(cls) -> Self:
```

Comment:
I guess class methods can't be properties?

Reply:
Interesting, it occurred to me but I didn't check, I can try.

Reply:
Python (tests) works, but `pylint` doesn't :roll_eyes: 
```
tests/timeseries/test_quantities.py:47:11: W0143: Comparing against a callable, did you omit the parenthesis? (comparison-with-callable)
```

And `mypy` explodes (again) :boom: (lately I found `mypy` is very unstable, it happens very often that I make a silly change and `mypy` with start throwing an `INTERNAL ERROR`. Sometimes if I run it outside `nox` it will give an error, but sometimes it doesn't. I guess we are taking it to the limit. :unamused: 

Running with `--show-traceback` at least I get which line is making it crash and then looking inside vim, I see the mypy error, so I guess what's killing it is processing all the files at once, and the `mypy` error is:

```
Only instance methods can be decorated with @property  [misc] (mypy error)
```

So weird, because again, Python does support it. OK, researching some more, it seems like it used to work but it was deprecated in 3.11:

> Chaining classmethod descriptors (introduced in bpo-19072) is now deprecated. It can no longer be used to wrap other descriptors such as property. The core design of this feature was flawed and caused a number of downstream problems.

https://github.com/python/mypy/issues/13746#issuecomment-1258864382

Soooooo.... No, we can't do that :P

Reply:
Yeah, I noticed that the mypy people seem to have their own ideas of what is and isn't possible, same with the percentage __mul__ stuff